addon/v1alpha1/types_clustermanagementaddon.go

```diff
@@ -175,10 +185,146 @@ type PlacementStrategy struct {
 	// User can override the configuration by updating the managedClusterAddon directly.
 	// +optional
 	Configs []AddOnConfig `json:"configs,omitempty"`
+	// The rollout strategy to apply addon configurations change.
+	// The rollout strategy only watches the addon configurations defined in ClusterManagementAddOn.
+	// +kubebuilder:default={type: UpdateAll}
+	// +optional
+	RolloutStrategy *RolloutStrategy `json:"rolloutStrategy,omitempty"`
```

Comment:
why pointer? I am starting to think it should not be a pointer, InstallStrategy should not be pointer either

Reply:
The `kubebuilder:default={type: UpdateAll}` won't work when RolloutStrategy is a structure, like this integration testing case shows https://github.com/open-cluster-management-io/api/pull/208/files#diff-0f15edf563bb9b75e4f6b5934e42dffcc3db1c80238ae2970cafce38a8959c54

Reply:
yeah, that is because when you use client to create the resource, this field is init by default. But when you use yaml without setting this field, it should still work

Reply:
Set both InstallStrategy and RolloutStrategy to structure, and modified some integration tests.